plugins/nominal-connection-checker/src/hierarchy_validation.js

```diff
@@ -228,6 +229,48 @@ function checkSupersDefined(hierarchyDef) {
   }
 }
 
+/**
+ * Checks the hierarchy def for any types which include the same type multiple
+ * times in their fulfills array.
```

Comment:
I think you could add that this logs an error for any types which includes the same type multiple times. 
That way it's clear from the start how this method handles finding the same type.